storage/diemdb/src/pruner/mod.rs

```diff
@@ -223,6 +202,51 @@ impl Worker {
         }
     }
 
+    /// Find out the first undeleted item in the stale node index.
+    ///
+    /// Seeking from the beginning (version 0) is potentially costly, we do it once upon worker
+    /// thread start, record the progress and seek from that position afterwards.
+    fn initialize(&mut self) {
+        loop {
+            match self.get_least_readable_version() {
+                Ok(least_readable_version) => {
+                    info!(
+                        least_readable_version = least_readable_version,
+                        "[state pruner worker] initialized."
+                    );
+                    self.target_least_readable_version = least_readable_version;
+                    self.record_progress(least_readable_version);
+                    return;
+                }
+                Err(e) => {
+                    error!(
+                        error = ?e,
+                        "[state pruner worker] Error on first seek. Retrying in 1 second.",
+                    );
+                    sleep(Duration::from_secs(1));
```

Comment:
Wonder if longer window is needed - would there be a flood of logs during the time a node restarts and works its way through finding the right postiion?

Reply:
No, it's seeking the local DB and failure is extremely unlikely. It's not something that prevents the node to run, hence a retry loop, otherwise I'd just panic!.